src/tests/test_backend.py

```diff
@@ -151,17 +151,20 @@ def test_cannot_delete_password(passwords, fake_provider):
 
 
 def test_retry_on_invalid_credentials(validating_provider):
-    username, password = validating_provider.get_credentials(SUPPORTED_HOST + "200")
-    assert password == False # credentials returned from single call with IsRetry=false
+     # No credentials returned when it can already authenticate without them
+    username, password = validating_provider.get_credentials("200" + SUPPORTED_HOST)
+    assert username == None and password == None
 
-    username, password = validating_provider.get_credentials(SUPPORTED_HOST + "401")
-    assert password == True # credentials returned from second call with IsRetry=true
+    # Credentials returned from first call with IsRetry=false
+    username, password = validating_provider.get_credentials("200" + SUPPORTED_HOST + "pypi/upload")
+    assert password == False
 
-    username, password = validating_provider.get_credentials(SUPPORTED_HOST + "403")
+    # Credentials returned from second call with IsRetry=true
+    username, password = validating_provider.get_credentials("401" + SUPPORTED_HOST)
     assert password == True
 
-    username, password = validating_provider.get_credentials(SUPPORTED_HOST + "500")
+    username, password = validating_provider.get_credentials("403" + SUPPORTED_HOST)
     assert password == True
 
-    username, password = validating_provider.get_credentials(SUPPORTED_HOST + "404")
```

Comment:
No need to test 404 anymore?

Reply:
Nope, this validation was wrong. It was meant to test the case where the credential provider is called again (this time wiht IsRetry=True), but it was actually not reaching that point and just returning "None" credentials, which is the case when the call can authenticate without credentials (public feed case).